x-pack/plugins/xpack_main/server/routes/api/v1/xpack_usage.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { wrap, serverTimeout as serverUnavailable } from 'boom';
+
+const getClusterUuid = async req => {
+  const { server } = req;
+  const { callWithRequest, } = server.plugins.elasticsearch.getCluster('data');
+  const { cluster_uuid: uuid } = await callWithRequest(req, 'info', { filterPath: 'cluster_uuid', });
+  return uuid;
+};
+
+/*
+ * @return {Object} data from usage stats collectors registered with Monitoring CollectorSet
+ */
+const getUsage = async req => {
+  const server = req.server;
+  const { collectorSet } = server.plugins.monitoring;
+  if (collectorSet === undefined) {
+    const error = new Error('CollectorSet from Monitoring plugin is not ready for collecting usage'); // moving kibana_monitoring lib to xpack_main will make this unnecessary
+    throw serverUnavailable(error);
+  }
+  return collectorSet.bulkFetchUsage();
+};
+
+export function xpackUsageRoute(server) {
+  server.route({
+    path: '/api/_xpack/usage',
```

Comment:
I like the usage of `_xpack` here because it's consistent with ES X-Pack, but I wonder how others might feel about this.

Reply:
pretty much every other API in Kibana has a convention of `/api/kibana/<featurename>/<sub>/`

```
src/core_plugins/kibana/server/routes/api/export/index.js:7:5:    path: '/api/kibana/dashboards/export',
src/core_plugins/kibana/server/routes/api/home/register_tutorials.js:4:5: path: '/api/kibana/home/tutorials',
src/core_plugins/kibana/server/routes/api/scroll_search/index.js:5:5:    path: '/api/kibana/legacy_scroll_start',
src/core_plugins/kibana/server/routes/api/search/count/register_count.js:6:5: path: '/api/kibana/{id}/_count',
src/core_plugins/kibana/server/routes/api/scroll_search/index.js:24:5:    path: '/api/kibana/legacy_scroll_continue',
src/core_plugins/kibana/server/routes/api/scripts/register_languages.js:3:5: path: '/api/kibana/scripts/languages',
src/core_plugins/kibana/server/routes/api/suggestions/register_value_suggestions.js:6:5: path: '/api/kibana/suggestions/values/{index}',
src/core_plugins/kibana/server/routes/api/management/saved_objects/scroll.js:19:5: path: '/api/kibana/management/saved_objects/scroll/export',
src/core_plugins/kibana/server/routes/api/management/saved_objects/scroll.js:54:5: path: '/api/kibana/management/saved_objects/scroll/counts',
src/core_plugins/kibana/server/routes/api/management/saved_objects/relationships.js:7:5: path: '/api/kibana/management/saved_objects/relationships/{type}/{id}',
```

The existing status API has a path of `/api/status` and the new stats API (6.3.0) has a path of `/api/stats`

I think it's good to be consistent with ES, as Kibana doesn't have a good foundation for conventions currently.